File: spring/src/main/java/io/github/wavesync/entity/FinalStat.java
package io.github.wavesync.entity;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;

@Entity
@Table(name = "final_stats")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class FinalStat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Integer hp;

    @Column(nullable = false)
    private Integer attack;

    @Column(nullable = false)
    private Integer defense;

    @Column(name = "energy_regen", nullable = false, precision = 5, scale = 2)
    private BigDecimal energyRegen;

    @Column(name = "critical_rate",nullable = false, precision = 5, scale = 2)
    private BigDecimal criticalRate;

    @Column(name = "critical_damage", nullable = false, precision = 5, scale = 2)
    private BigDecimal criticalDamage;

    @Column(name = "resonance_skill_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal resonanceSkillDamageBonus;

    @Column(name = "basic_attack_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal basicAttackDamageBonus;

    @Column(name = "heavy_attack_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal heavyAttackDamageBonus;

    @Column(name = "resonance_liberation_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal resonanceLiberationDamageBonus;

    @Column(name = "glacio_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal glacioDamageBonus;

    @Column(name = "fusion_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal fusionDamageBonus;

    @Column(name = "conducto_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal conductoDamageBonus;

    @Column(name = "aero_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal aeroDamageBonus;

    @Column(name = "spectro_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal spectroDamageBonus;

    @Column(name = "havoc_damage_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal havocDamageBonus;

    @Column(name = "healing_bonus", nullable = false, precision = 5, scale = 2)
    private BigDecimal healingBonus;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_resonator_id", nullable = false)
    private UserResonator userResonator;
}

File: fastapi/app/models/final_stat.py
from sqlalchemy import BigInteger, Column, ForeignKey, Integer, Numeric
from sqlalchemy.orm import relationship

from app.db.base import Base


class FinalStat(Base):
    """공명자 최종 스탯.

    final_stats.user_resonator_id엔 DB상 UNIQUE 제약이 없어 unique=True를 넣지 않는다.
    """

    __tablename__ = "final_stats"

    id = Column(BigInteger, primary_key=True)

    hp = Column(Integer, nullable=False)
    attack = Column(Integer, nullable=False)
    defense = Column(Integer, nullable=False)

    energy_regen = Column(Numeric(5, 1), nullable=False)
    critical_rate = Column(Numeric(5, 1), nullable=False)
    critical_damage = Column(Numeric(5, 1), nullable=False)

    resonance_skill_damage_bonus = Column(Numeric(5, 1), nullable=False)
    basic_attack_damage_bonus = Column(Numeric(5, 1), nullable=False)
    heavy_attack_damage_bonus = Column(Numeric(5, 1), nullable=False)
    resonance_liberation_damage_bonus = Column(Numeric(5, 1), nullable=False)

    glacio_damage_bonus = Column(Numeric(5, 1), nullable=False)
    fusion_damage_bonus = Column(Numeric(5, 1), nullable=False)
    conducto_damage_bonus = Column(Numeric(5, 1), nullable=False)
    aero_damage_bonus = Column(Numeric(5, 1), nullable=False)
    spectra_damage_bonus = Column(Numeric(5, 1), nullable=False)
    havoc_damage_bonus = Column(Numeric(5, 1), nullable=False)
    healing_bonus = Column(Numeric(5, 1), nullable=False)

    user_resonator_id = Column(BigInteger, ForeignKey("user_resonators.id"), nullable=False)
    user_resonator = relationship("UserResonator", back_populates="final_stat")
